demos/python_demos/object_detection_demo_ssd_async/object_detection_demo_ssd_async.py

```diff
@@ -15,239 +15,593 @@
-    else:
-        labels_map = None
 
-    cur_request_id = 0
-    next_request_id = 1
+    log.info('Loading network...')
+    completed_request_results = {}
+    modes = cycle(Modes)
+    prev_mode = mode = next(modes)
+    log.info('Using {} mode'.format(mode.name))
+    mode_info = {mode: ModeInfo()}
+    exceptions = []
+
+    exec_nets = {
+        Modes.USER_SPECIFIED:
+            Detector(ie, args.model, device=args.device, plugin_config=config_user_specified,
+                     results=completed_request_results, max_num_requests=args.num_infer_requests,
+                     labels_map=labels_map, keep_aspect_ratio_resize=args.keep_aspect_ratio,
+                     caught_exceptions=exceptions),
+        Modes.MIN_LATENCY:
+            Detector(ie, args.model, device=args.device.split(':')[-1].split(',')[0], plugin_config=config_min_latency,
+                     results=completed_request_results, max_num_requests=1,
+                     labels_map=labels_map, keep_aspect_ratio_resize=args.keep_aspect_ratio,
+                     caught_exceptions=exceptions)
+    }
+
+    try:
+        input_stream = int(args.input)
+    except ValueError:
+        input_stream = args.input
+    cap = cv2.VideoCapture(input_stream)
+    wait_key_time = 1
 
-    log.info("Starting inference in async mode...")
-    is_async_mode = True
-    render_time = 0
-    if is_async_mode:
-        ret, frame = cap.read()
-        frame_h, frame_w = frame.shape[:2]
+    next_frame_id = 0
+    next_frame_id_to_show = 0
+    input_repeats = 0
+  
+    log.info('Starting inference...')
+    print("To close the application, press 'CTRL+C' here or switch to the output window and press ESC key")
+    print("To switch between min_latency/user_specified modes, press TAB key in the output window")
 
-    presenter = monitors.Presenter(args.utilization_monitors, 45,
+    palette = ColorPalette(len(labels_map) if labels_map is not None else 100)
+    presenter = monitors.Presenter(args.utilization_monitors, 55,
         (round(cap.get(cv2.CAP_PROP_FRAME_WIDTH) / 4), round(cap.get(cv2.CAP_PROP_FRAME_HEIGHT) / 8)))
 
-    print("To close the application, press 'CTRL+C' here or switch to the output window and press ESC key")
-    print("To switch between sync/async modes, press TAB key in the output window")
+    while (cap.isOpened() \
+           or completed_request_results \
+           or len(exec_nets[mode].empty_requests) < len(exec_nets[mode].requests)) \
+          and not exceptions:
+        if next_frame_id_to_show in completed_request_results:
+            frame_meta, raw_outputs = completed_request_results.pop(next_frame_id_to_show)
+            objects = exec_nets[mode].postprocess(raw_outputs, frame_meta)
+
+            frame = frame_meta['frame']
+            start_time = frame_meta['start_time']
+
+            if len(objects) and args.raw_output_message:
+                log.info(' Class ID | Confidence | XMIN | YMIN | XMAX | YMAX | COLOR ')
```

Comment:
```suggestion
                log.info(' Class ID | Confidence | XMIN | YMIN | XMAX | YMAX | COLOR ')
```

Why does the color need to be here? That's not an intrinsic property of the object.

Reply:
Just preserved the format as used in yolov3 demo. I could only guess that it could be useful to map printed detections and the ones drawn, when the caption is not readable. Don't mind to remove it here, though.